db/views/reporting_overdue_patients_v01.sql

```diff
@@ -0,0 +1,89 @@
+SELECT DISTINCT ON (reporting_patient_states.month_date, reporting_patient_states.patient_id)
+  reporting_patient_states.month_date AS month_date,
+  reporting_patient_states.patient_id AS patient_id,
+  reporting_patient_states.assigned_facility_id AS assigned_facility_id,
+  reporting_patient_states.assigned_facility_region_id AS assigned_facility_region_id,
+  appointments.id AS previous_appointment_id,
+  appointments.device_created_at AT TIME ZONE 'UTC' AT TIME ZONE 'UTC' AS previous_appointment_date,
+  appointments.scheduled_date AT TIME ZONE 'UTC' AT TIME ZONE 'UTC' AS previous_appointment_scheduled_date,
+  visits.visited_at_after_appointment AT TIME ZONE 'UTC' AT TIME ZONE 'UTC' AS visited_at_after_appointment,
+  next_call_results.device_created_at AT TIME ZONE 'UTC' AT TIME ZONE 'UTC' AS next_called_at,
+  previous_call_results.device_created_at AT TIME ZONE 'UTC' AT TIME ZONE 'UTC' AS previous_called_at,
+  reporting_patient_states.hypertension AS hypertension,
+  reporting_patient_states.diabetes AS diabetes,
+  next_call_results.result_type AS next_call_result_type,
+  next_call_results.remove_reason AS next_call_remove_from_overdue_list_reason,
+  previous_call_results.result_type AS previous_call_result_type,
+  previous_call_results.remove_reason AS previous_call_remove_from_overdue_list_reason,
+  CASE
+    WHEN (appointments.scheduled_date >= reporting_patient_states.month_date) THEN 'no'
+    WHEN (appointments.scheduled_date < reporting_patient_states.month_date
+	AND visits.visited_at_after_appointment < reporting_patient_states.month_date) THEN 'no'
+    ELSE 'yes'
+  END AS is_overdue,
+  CASE
+    WHEN next_call_results.device_created_at IS NULL THEN 'no'
+    ELSE 'yes'
+  END AS has_called,
+  CASE
+    WHEN visited_at_after_appointment IS NULL OR next_call_results.device_created_at IS NULL THEN 'no'
+    WHEN visited_at_after_appointment > next_call_results.device_created_at + interval '15 days' THEN 'no'
+    ELSE 'yes'
+  END AS has_visited_following_call,
+  CASE
+    WHEN reporting_patient_states.htn_care_state = 'lost_to_follow_up' THEN 'yes'
+    ELSE 'no'
+  END AS ltfu,
+  CASE
+    WHEN reporting_patient_states.htn_care_state = 'under_care' THEN 'yes'
+    ELSE 'no'
+  END AS under_care,
+  CASE
+    WHEN patient_phone_numbers.number IS NULL THEN 'no'
+    ELSE 'yes'
+  END AS has_phone,
+  CASE
+    WHEN previous_call_results.result_type = 'removed_from_overdue_list' THEN 'yes'
+    ELSE 'no'
+  END AS removed_from_overdue_list
+FROM reporting_patient_states
+  LEFT JOIN appointments ON reporting_patient_states.patient_id = appointments.patient_id
+  AND appointments.device_created_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE')) < reporting_patient_states.month_date
```

Comment:
[non-blocking] A different way to write this could be
- get latest appointment per patient per month in a WITH clause
- left outer join on appointments.month = reporting_patient_states.month - 1 

Suspecting this might be more performant since the join happens on a smaller table (instead of all of appointments which is pretty large).

Reply:
Interesting, we'll check this out separately. But I suspect the query planner will do this optimisation.